### ISSUE TITLE
CI fails on PRs from external forks

### ISSUE THREAD

**Issue by MantisClone** [RequestNetwork/requestNetwork#1029]
# Problem
The `payment-detection` tests rely on the `EXPLORER_API_KEY_*` secrets. Secrets are not available to PRs from external forks.

# Example
https://app.circleci.com/pipelines/github/RequestNetwork/requestNetwork/7596/workflows/7c85970e-2bc8-4c54-b8dc-1e1551cea23e/jobs/124290

# Solution
Skip tests that rely on secrets in PRs from external forks.

# Details
> The particular variable of interest to us is `CIRCLE_PR_NUMBER`, documented as “the number of the associated GitHub or Bitbucket pull request. Only available on forked PRs.” If `CIRCLE_PR_NUMBER` exists, then we know we are running a build for a forked PR that does not have access to secrets.
>
> To express this condition in shell syntax, we use the -n (non-zero length) test. The condition will look like this:

    if [ -n "$CIRCLE_PR_NUMBER" ]; then
      # mark this job successful and stop processing
    fi

# Reference
* https://circleci.com/blog/managing-secrets-when-you-have-pull-requests-from-outside-contributors/

**Comment by MantisClone** [comment]
After #1030 was merged, CI on PRs from external forks is still failing. :cry:

Example: https://app.circleci.com/pipelines/github/RequestNetwork/requestNetwork/7632/workflows/32be1005-bf67-4dcf-b07c-981d60df0d8f/jobs/124849

```console
yarn publish-coverage
yarn run v1.22.18
$ mkdir -p ./coverage && node scripts/collect.js && nyc merge ./coverage ./coverage/coverage.json && node scripts/remap.js && cat coverage/lcov.info | coveralls
coverage files in ./coverage merged into ./coverage/coverage.json
(node:817) Warning: Accessing non-existent property 'VERSION' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)
remapping finished.

/home/circleci/repo/node_modules/coveralls/bin/coveralls.js:19
      throw err;
      ^
Bad response: 422 {"message":"Couldn't find a repository matching this job.","error":true}
(Use `node --trace-uncaught ...` to show where the exception was thrown)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

Exited with code exit status 1

```

**Comment by MantisClone** [comment]
Just to close the loop, @alexandre-abrioux fixed the issue by making the `publish-coverage` step optional instead of required.
